src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToAggregationExpressionTranslators/BinaryExpressionToAggregationExpressionTranslator.cs

```diff
@@ -30,13 +31,10 @@ public static AggregationExpression Translate(TranslationContext context, Binary
                 return StringGetCharsComparisonExpressionToAggregationExpressionTranslator.Translate(context, expression, getCharsExpression);
             }
 
-            var leftExpression = expression.Left;
-            var rightExpression = expression.Right;
-            if (IsArithmeticExpression(expression))
-            {
-                leftExpression = ConvertHelper.RemoveWideningConvert(leftExpression);
-                rightExpression = ConvertHelper.RemoveWideningConvert(rightExpression);
-            }
+            var leftExpression = ConvertHelper.RemoveWideningConvert(
+                                 ConvertHelper.RemoveEnumConvert(expression.Left));
+            var rightExpression = ConvertHelper.RemoveWideningConvert(
+                                  ConvertHelper.RemoveEnumConvert(expression.Right));
```

Comment:
We had a big problem in LINQ2 where we were promiscuously removing Converts without verifying that it was really OK to do so.

I want us to only remove a `Convert` when we have carefully verified that we should.

Replace 34-37 with:

    var leftExpression = expression.Left;
    var rightExpression = expression.Right;
    if (IsArithmeticExpression(expression))
    {
        leftExpression = ConvertHelper.RemoveWideningConvert(leftExpression);
        rightExpression = ConvertHelper.RemoveWideningConvert(rightExpression);
    }
    if (IsEnumComparisonExpression(expression))
    {
        leftExpression = ConvertHelper.RemoveConvertToEnumUnderlyingType(leftExpression);
        rightExpression = ConvertHelper.RemoveConvertToEnumUnderlyingType(rightExpression);
    }

Reply:
`IsArithmeticExpression` is an already existing helper method.

Reply:
`IsEnumComparisonExpression` would be:

    private static bool IsEnumComparisonExpression(BinaryExpression expression)
    {
        return
            IsComparisonOperator(expression.NodeType) &&
            (IsConvertToEnumUnderlyingType(expression.Left) || IsConvertToEnumUnderlyingType(expression.Right));

        static bool IsConvertToEnumUnderlyingType(Expression expression)
        {
            if (expression.NodeType == ExpressionType.Convert)
            {
                var convertExpression = (UnaryExpression)expression;
                var sourceType = convertExpression.Operand.Type;
                var targetType = convertExpression.Type;
                return sourceType.IsEnum() && targetType == Enum.GetUnderlyingType(sourceType);
            }

            return false;
        }
    }

Reply:
And `IsComparisonOperator` would be:

    private static bool IsComparisonOperator(ExpressionType nodeType)
    {
        return nodeType switch
        {
            ExpressionType.Equal => true,
            ExpressionType.GreaterThan => true,
            ExpressionType.GreaterThanOrEqual => true,
            ExpressionType.LessThan => true,
            ExpressionType.LessThanOrEqual => true,
            ExpressionType.NotEqual => true,
            _ => false
        };
    }